File: mockmate-backend/src/main/resources/runners/python/runner_binary_tree.py
import sys, math, heapq, bisect, re
from typing import List, Optional, Dict, Set, Tuple, Any
from collections import deque, Counter, defaultdict, OrderedDict
from collections import deque

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def build_tree(values):
    if not values or values[0] == 'null':
        return None
    try:
        root = TreeNode(int(values[0]))
    except ValueError:
        raise ValueError(f"Invalid value at index 0: \'{values[0]}\' — expected integer or \'null\'")
    queue = deque([root])
    i = 1
    while queue and i < len(values):
        node = queue.popleft()
        if i < len(values) and values[i] != 'null':
            try:
                node.left = TreeNode(int(values[i]))
            except ValueError:
                raise ValueError(f"Invalid value at index {i}: \'{values[i]}\' — expected integer or \'null\'")
            queue.append(node.left)
        i += 1
        if i < len(values) and values[i] != 'null':
            try:
                node.right = TreeNode(int(values[i]))
            except ValueError:
                raise ValueError(f"Invalid value at index {i}: \'{values[i]}\' — expected integer or \'null\'")
            queue.append(node.right)
        i += 1
    return root

def serialize_tree(root):
    if not root:
        return 'null'
    result = []
    queue = deque([root])
    while queue:
        node = queue.popleft()
        if node:
            result.append(str(node.val))
            queue.append(node.left)
            queue.append(node.right)
        else:
            result.append('null')
    while result and result[-1] == 'null':
        result.pop()
    return ' '.join(result)

{{USER_CODE}}

lines = sys.stdin.read().strip().split('\n')
values = lines[0].strip().split()
root = build_tree(values)
sol = Solution()
result = sol.{{methodSignature}}(root)
if isinstance(result, TreeNode):
    print(serialize_tree(result))
elif isinstance(result, bool):
    print(str(result).lower())
else:
    print(result)

File: mockmate-backend/src/main/resources/runners/java/runner_binary_tree.java
import java.util.*;

public class Main {

    static class TreeNode {
        int val;
        TreeNode left, right;

        TreeNode(int val) {
            this.val = val;
        }
    }

    static TreeNode buildTree(String input) {
        if (input.isEmpty())
            return null;
        String[] parts = input.trim().split("\\s+");
        if (parts.length == 0 || parts[0].isEmpty() || parts[0].equals("null"))
            return null;

        TreeNode root = new TreeNode(Integer.parseInt(parts[0]));
        Queue<TreeNode> queue = new LinkedList<>();
        queue.offer(root);
        int i = 1;
        while (!queue.isEmpty() && i < parts.length) {
            TreeNode node = queue.poll();
            if (i < parts.length && !parts[i].equals("null")) {
                node.left = new TreeNode(Integer.parseInt(parts[i]));
                queue.offer(node.left);
            }
            i++;
            if (i < parts.length && !parts[i].equals("null")) {
                node.right = new TreeNode(Integer.parseInt(parts[i]));
                queue.offer(node.right);
            }
            i++;
        }
        return root;
    }

    static String serializeTree(TreeNode root) {
        if (root == null)
            return "null";
        StringBuilder sb = new StringBuilder();
        Queue<TreeNode> queue = new LinkedList<>();
        queue.offer(root);
        while (!queue.isEmpty()) {
            TreeNode node = queue.poll();
            if (sb.length() > 0)
                sb.append(" ");
            if (node == null) {
                sb.append("null");
            } else {
                sb.append(node.val);
                queue.offer(node.left);
                queue.offer(node.right);
            }
        }
        // Trim trailing nulls
        String result = sb.toString();
        while (result.endsWith(" null")) {
            result = result.substring(0, result.length() - 5);
        }
        return result;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String input = sc.hasNextLine() ? sc.nextLine().trim() : "";
        TreeNode root = buildTree(input);

        Solution sol = new Solution();
        Object result = sol.{{methodSignature}}(root);

        if (result instanceof TreeNode) {
            System.out.println(serializeTree((TreeNode) result));
        } else {
            System.out.println(result);
        }
    }
}

{{USER_CODE}}
